kas/anchor.go

```diff
@@ -66,6 +68,86 @@ func NewKASAnchor(kasConfig *KASConfig, db AnchorDB, bc BlockChain) *Anchor {
 	}
 }
 
+// AnchorPeriodicBlock periodically anchor blocks to KAS.
+// if given block is invalid, it does nothing.
+func (anchor *Anchor) AnchorPeriodicBlock(block *types.Block) {
+	if !anchor.kasConfig.Anchor {
+		return
+	}
+
+	if block == nil {
+		logger.Error("KAS Anchor : failed to get a block form the event")
+		return
+	}
+
+	if block.NumberU64()%anchor.kasConfig.AnchorPeriod != 0 {
+		return
+	}
+
+	anchor.anchorBlock(block)
+}
+
+// AnchorByBlockNum anchors the block of given block number to KAS by API.
+func (anchor *Anchor) AnchorByBlockNum(blkNum uint64) error {
+	block := anchor.bc.GetBlockByNumber(blkNum)
+	if block == nil {
+		return errNotFoundBlock
+	}
+
+	return anchor.anchorBlock(block)
+}
+
+// blockToAnchoringDataInternalType0 makes AnchoringDataInternalType0 from the given block.
+// TxCount is the total number of transactions during N-P+1 ~ N blocks. (N: given block, P: anchoring period)
+func (anchor *Anchor) blockToAnchoringDataInternalType0(block *types.Block) *types.AnchoringDataInternalType0 {
```

Comment:
Is this function called only when `block.NumberU64() % anchor.kasConfig.AnchorPeriod == 0`?

Reply:
`chainEvent` calls this func with the right periodic block number.
But API can calls this func with any block number.

Reply:
As we discussed off-line, consider changing the semantic of this function to returning last N block number.

Reply:
reflected